controllers/common/common_status_update.go

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) 2023 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+	appsv1 "open-cluster-management.io/multicloud-operators-subscription/pkg/apis/apps/placementrule/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	policiesv1beta1 "open-cluster-management.io/governance-policy-propagator/api/v1beta1"
+)
+
+// RootStatusUpdat update Root policy status with bound decisions and placements
+func RootStatusUpdate(c client.Client, rootPolicy *policiesv1.Policy) error {
```

Comment:
Is this function actually needed to be split off? I only see it used in the root policy status sync controller.

Reply:
yes but like GetAllClusterDecisions, GetPolicyPlacementDecisions, etc used in rootpolicycontroller

Reply:
oh you mean only RootStatusUpdate can be moved
 to as a RootPolicyStatusReconciler? OH that  is good idea

Reply:
Yeah, you can just keep the code in RootPolicyStatusReconcile as is I think since RootStatusUpdate isn't reused in your new code.